### DISCUSSION TITLE
Compiling HelloDrone c++ on Ubuntu

### DISCUSSION THREAD

**Discussion by mmuetert** [microsoft/AirSim#3529] [Support Q&A]
I am trying to compile HelloDrone to get a feel for the C++ APIs. I am working in Ubuntu 18.04. I went to the folder AirSim/cmake/HelloDrone, created a build directory, changed to the build directory and ran cmake. It seemed to run alright, generating a bunch of files in build. I then tried to run make, but received the following  errors:

/usr/bin/ld: cannot find -lAirLib
/usr/bin/ld: cannot find -lrpc

I know these mean the system can't find the libraries, but I don't know how to fix it. Can someone help? I am new to AirSim and C++, so apologies if this is a trivial problem.

**Comment by ahmed-elsaharti** [comment]
Hi @mmuetert , AFAIK when you [build AirSim](https://microsoft.github.io/AirSim/build_linux/) the script builds all the C++ projects included and generates executables under  `AirSim/build_release/output/bin` (or at least thats what happened when I tested this)

**Reply by mmuetert** [reply]
Thank you for the response. I was able to find the code and execute it. That still leaves me with a question, though. If I were to write my own C++ program, created a CMakeList.txt file for it, and ran cmake and make, wouldn't I still run into the same issue that the compiler can't find those libraries?

Thanks!

**Reply by ahmed-elsaharti** [reply]
According to [the guide](https://microsoft.github.io/AirSim/apis_cpp/) all you need to do is setup the paths for the Include and Lib folders in cmake. I haven't tried it myself though.

**Reply by mmuetert** [reply]
I have read the guide and I can't figure it out how to properly make the cmakelist.txt file or where I need to to put my custom cpp files. Can we get more in depth tutorial about building custom C++ control files in Linux? the current documentation dedicates one sentence to Linux. I have tried reading through the cmake documentation to build the cmakelist files, but the files that come with AirSim seem to use different syntax than the cmake documentation/

thanks.

**Reply by rajat2004** [reply]
Maybe have a look at https://github.com/madratman/airsim_standalone, it might help in figuring out the problems

Did you run `build.sh` first before compiling your files, the error might be occurring since `build.sh` will first compile the static AirLib, etc libraries
